joshua/joshua_agent.py

```diff
@@ -519,6 +523,30 @@ def _run_ensemble(
 
         #    print('{} Running ensemble in dir: {}'.format(threading.current_thread().name, work_dir),file=getFileHandle())
 
+        # Check and set the Pod label when run inside a k8s pod
+        # env['HOSTNAME'] contains the pod name in k8s
+        namespace = ""
+        k8s_namespace_file = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+        if env['HOSTNAME'] and os.path.isfile(k8s_namespace_file):
+            pod = env['HOSTNAME']
+            with open(k8s_namespace_file, 'r') as f:
+                namespace = f.read()
+
+            # Get the cluster config
+            config.load_incluster_config()
+            v1 = client.CoreV1Api()
+
+            # If last_test label is set to true, exit right away
+            pod_desc = v1.read_namespaced_pod(pod, namespace)
+            try:
+                if pod_desc.metadata.labels['last_test']  == "true":
+                    raise JoshuaError("New agent scaler marked this pod to stop. Exiting.")
+            except KeyError:
+                pass
+
+            # Update the ensemble label
+            v1.patch_namespaced_pod(pod, namespace, {"metadata":{"labels":{"ensemble":ensemble}}})
```

Comment:
I don't have enough context about joshua, but should we check if the label is already correct instead of doing the patch always?